central/resourcecollection/datastore/datastore_impl.go

```diff
@@ -321,6 +321,24 @@ func (ds *datastoreImpl) addCollectionWorkflow(ctx context.Context, collection *
 	return nil
 }
 
+func (ds *datastoreImpl) AddTestCollection(ctx context.Context) (string, error) {
```

Comment:
This shouldn't be part of the datastore interface especially as it's only used in tests.

Instead move to somewhere like datastore/test_utils.go. Take in the datastore as a param.

Also for test methods the best practice is to taking in a testing.TB (testing.T or testing.B) so that it ensures it's only ever called in a test.

```
func AddTestCollection(_ testing.TB, ctx context.Context, ds *Datastore) (string, error)
```